pkg/ebpftracer/types/protocol.go

```diff
@@ -19,49 +16,32 @@ const (
 	SendVfsRead
 )
 
-// PLEASE NOTE, YOU MUST UPDATE THE DECODER IF ANY CHANGE TO THIS STRUCT IS DONE.
-type SignalContext struct {
-	EventID events.ID // uint32
-}
-
-func (SignalContext) GetSizeBytes() int {
-	return 4
-}
-
 // PLEASE NOTE, YOU MUST UPDATE THE DECODER IF ANY CHANGE TO THIS STRUCT IS DONE.
 
 // EventContext struct contains common metadata that is collected for all types of events
 // it is used to unmarshal binary data and therefore should match (bit by bit) to the `context_t` struct in the ebpf code.
-// NOTE: Integers want to be aligned in memory, so if changing the format of this struct
-// keep the 1-byte 'Argnum' as the final parameter before the padding (if padding is needed).
 type EventContext struct {
 	Ts        uint64
-	StartTime uint64
-	CgroupID  uint64
-	Pid       uint32
-	Tid       uint32
-	Ppid      uint32
-	HostPid   uint32
-	HostTid   uint32
-	HostPpid  uint32
+	StartTime uint64 `sourceField:"Task.StartTime"`
+	CgroupID  uint64 `sourceField:"Task.CgroupId"`
+	Pid       uint32 `sourceField:"Task.Pid"`
+	Tid       uint32 `sourceField:"Task.Tid"`
+	Ppid      uint32 `sourceField:"Task.Ppid"`
+	HostPid   uint32 `sourceField:"Task.HostPid"`
+	HostTid   uint32 `sourceField:"Task.HostTid"`
+	HostPpid  uint32 `sourceField:"Task.HostPpid"`
 	// PID translated to PIDNS the container runtime is running in
-	NodeHostPid     uint32
-	Uid             uint32
-	MntID           uint32
-	PidID           uint32
-	Comm            [16]byte
-	UtsName         [16]byte
-	Flags           uint32
-	_               [4]byte   // padding
```

Comment:
is there any impact to removing this padding?

Reply:
Before this was more or less an exact representation of the thing we sent over the perfbuf to userspace. The only downside it would have was that tests would fail. 

Now I completely redid how we are doing this, by leveraging the auto generated type from ciliums bpf2go.

Long story short, padding is no longer required here.